File: apis_c#/AthleteApi/AthleteApi/Models/TournamentParticipation.cs
using Swashbuckle.AspNetCore.Annotations;

namespace AthleteApi.Models
{
    [SwaggerSchema(Description = "Representa la participación de un atleta en un torneo")]
    public class TournamentParticipation
    {
        [SwaggerSchema("Identificador único de la participación")]
        public int Id { get; set; }

        [SwaggerSchema("Identificador del atleta")]
        public int AthleteId { get; set; }

        [SwaggerSchema("Identificador del torneo")]
        public int TournamentId { get; set; }

        [SwaggerSchema("Identificador de la categoría de peso")]
        public int WeightCategoryId { get; set; }
    }
}

File: apis_py/app/schemas/tournament_participation.py
from pydantic import BaseModel, Field

class TournamentParticipationCreate(BaseModel):
    """
    Esquema de creación para la entidad TournamentParticipation.

    Atributos:
        athlete_id (int): Identificador del atleta. Ejemplo: 1.
        tournament_id (int): Identificador del torneo. Ejemplo: 1.
        weight_category_id (int): Identificador de la categoría de peso. Ejemplo: 1.
    """
    athlete_id: int = Field(..., title="ID del atleta", description="Identificador del atleta", example=1)
    tournament_id: int = Field(..., title="ID del torneo", description="Identificador del torneo", example=1)
    weight_category_id: int = Field(..., title="ID de la categoría de peso", description="Identificador de la categoría de peso", example=1)

class TournamentParticipationResponse(BaseModel):
    """
    Esquema de respuesta para la entidad TournamentParticipation.
    
    Atributos:
        id (int): Identificador único de la participación en el torneo. Ejemplo: 1.
        athlete_id (int): Identificador del atleta. Ejemplo: 1.
        tournament_id (int): Identificador del torneo. Ejemplo: 1.
        weight_category_id (int): Identificador de la categoría de peso. Ejemplo: 1.
    """
    id: int = Field(..., title="ID de la participación", description="Identificador único de la participación en el torneo", example=1)
    athlete_id: int = Field(..., title="ID del atleta", description="Identificador del atleta", example=1)
    tournament_id: int = Field(..., title="ID del torneo", description="Identificador del torneo", example=1)
    weight_category_id: int = Field(..., title="ID de la categoría de peso", description="Identificador de la categoría de peso", example=1)

    class Config:
        from_attributes = True  # Permite usar atributos de la clase para serialización